spec/lib/stellar/sep10_spec.rb

```diff
@@ -519,6 +519,38 @@
         /signers with weight 1 do not meet threshold 2./
       )
     end
+
+    it "raises an error for duplicate signatures" do
+      server_kp = Stellar::KeyPair.random
+      client_kp = Stellar::KeyPair.random
+
+      challenge_envelope = Stellar::TransactionEnvelope.from_xdr(
+        sep10.build_challenge_tx(
+          server: server_kp,
+          client: client_kp,
+          anchor_name: "SDF",
+          timeout: 600,
+        ), 
+        "base64"
+      )
+      challenge_envelope.signatures += [
+        challenge_envelope.tx.sign_decorated(client_kp),
+        challenge_envelope.tx.sign_decorated(client_kp)
+      ]
+      challenge = challenge_envelope.to_xdr(:base64)
+
+      expect {
+        sep10.verify_challenge_tx_threshold(
+          challenge_xdr: challenge,
+          server: server_kp,
+          signers: Set[{'key' => client_kp.address, 'weight' => 1}],
```

Comment:
Could you add a second version of this test that has the `signers` be an `Array` that contains the client kp twice?

Reply:
Can't pass Arrays, but I'll add a second signer with the same address but different weight. This will be deduplicated before passing to `verify_challenge_tx_signers` though, which builds the set of addresses.